packages/web3-providers/src/rss3/index.ts

```diff
@@ -0,0 +1,65 @@
+import urlcat from 'urlcat'
+import RSS3 from 'rss3-next'
+import { RSS3_ENDPOINT } from './constants'
+import { RSS3BaseAPI } from '../types'
+import { fetchJSON } from '../helpers'
+
+export class RSS3API implements RSS3BaseAPI.Provider {
+    createRSS3(
+        address: string,
+        sign: (message: string) => Promise<string> = () => {
+            throw new Error('Not supported.')
+        },
+    ): RSS3 {
+        return new RSS3({
+            endpoint: RSS3_ENDPOINT,
+            address,
+            sign,
+        })
+    }
+    async getFileData<T>(rss3: RSS3, address: string, key: string) {
+        const file = await rss3.files.get(address)
+        if (!file) throw new Error('The account was not found.')
+        const descriptor = Object.getOwnPropertyDescriptor(file, key)
+        return descriptor?.value as T | undefined
+    }
+    async setFileData<T>(rss3: RSS3, address: string, key: string, data: T): Promise<T> {
+        const file = await rss3.files.get(address)
+        if (!file) throw new Error('The account was not found.')
+        const descriptor = Object.getOwnPropertyDescriptor(file, key)
+        const value = {
+            ...(descriptor?.value as T | undefined),
+            ...data,
+        }
+        rss3.files.set(Object.assign(file, { [key]: value }))
+        await rss3.files.sync()
+        return value as T
+    }
+    async getDonations(address: string) {
+        const url = urlcat('https://hub.pass3.me/assets/list', {
```

Comment:
extract domain as constants

```suggestion
        const url = urlcat(HUB_API, '/assets/list', {
```